### ISSUE TITLE
Add composite dead-wood ordering and learn which neurons are most pruneable

### ISSUE THREAD

**Issue by nleck** [stSoftwareAU/NEAT-AI-Ockham#107]
## Goal

Order Ockham's sweep by the probability that a candidate will become a scorer-confirmed pruning win, rather than relying on one heuristic at a time.

The current implementation already exposes several useful signals independently: activation variance, mean absolute activation, activation range, outgoing contribution, fan-out, growth saving, squash type and historical learnings. Combine them into a better dead-wood priority model.

## Phase 1 — hand-built composite ordering

Add a composite metric using available features such as:

- activation variance / RMS / mean absolute activation;
- outgoing weight magnitude;
- fan-in / fan-out;
- downstream sensitivity;
- direct and cascade growth saving;
- squash type;
- topology depth/position where cheaply available;
- historical winner/failure evidence from prior epochs.

The metric should optimise expected scorer-verified pruning economics, e.g.:

```text
expected_pruning_value = P(candidate survives full scorer) × expected growth saving / expected evaluation cost
```

The exact formula should be benchmark-driven.

## Phase 2 — learn the ordering from Ockham history

Persist feature vectors with outcomes:

- sample delta;
- full delta;
- accepted/rejected;
- structure/growth units removed;
- scorer time.

Use this history to fit a very small transparent model such as a shallow decision tree or logistic model that predicts `P(full scorer accepts)` or expected pruning value.

This model **must only rank candidates**. It must never decide that a neuron can be removed without scorer verification.

## Requirements

- Random ordering remains available as control.
- New strategy must still visit every eligible neuron eventually.
- Reserve configurable random exploration so the learned model cannot permanently starve unusual candidates.
- Model/training data must be versioned and reproducible.
- Historical evidence from old corpus epochs is a prior, not current truth.
- Report comparison against individual existing orderings.

## Acceptance criteria

- Composite ordering implemented independently of learned ordering.
- Outcome telemetry is sufficient to train/evaluate the learned model offline.
- Learned strategy can be enabled experimentally without changing scorer acceptance semantics.
- Benchmark reports time-to-first-cut, confirmed cuts/hour, growth-units removed/hour and missed-winner rate.
- Promote only if it beats the random and best hand-written controls on real runs.

## Principle

Use intelligence to choose **what to test first**. Trust only the scorer to decide what survives. 🪒

**Comment by stservice** [comment]
✅ **Vibe Coder released this claim** — host `vibe-coder-2667`, finished 03:56 UTC. Raised #118 — https://github.com/stSoftwareAU/NEAT-AI-Ockham/pull/118

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI-Ockham/pull/118 has been created to address this issue.

**Comment by stservice** [comment]
## Issue run model stats

- **Requested model:** `opus`
- **Served model(s):** `claude-opus-5`
- **Effort:** `high`
- **Issue invocations:** 1
- **Tokens:** input 265 · output 163,831 · cache write 341,861 · cache read 35,591,737
- **Prompt cache:** 99.0% (read 35,591,737 · write 341,861 · uncached 265)
- **Turns:** 151
- **Duration:** 49m 29s
- **Estimated cost (USD, estimate only):** ~$24.03
  - `claude-opus-5`: $24.03 — input $0.0013 · output $4.10 · cache write $2.14 · cache read $17.80
- **Degraded:** no

_Estimate only — this block covers the run that posted it. The issue total sums the run-stats comments visible on this issue; runs that reported no figures are not included._
